modules/payment/commerce_payment.post_update.php

```diff
@@ -0,0 +1,82 @@
+<?php
+
+/**
+ * @file
+ * Post update functions for Payment.
+ */
+
+/**
+ * Update user Remote ID field data using payments gateways instead of module.
+ *
+ * @see https://www.drupal.org/node/2861181
+ */
+function commerce_payment_post_update_1(&$sandbox) {
+  // Get the user ids that have 'commerce_remote_id' field not empty.
+  // First check is any user has this field set.
+  if (\Drupal::entityQuery('user')->exists('commerce_remote_id')->count()->execute() == 0) {
+    return t('No user Remote ID field data to update.');
+  }
+  $result = \Drupal::entityQuery('user')->exists('commerce_remote_id')->execute();
+
+  // Use the sandbox to store the information needed to track progression.
+  if (!isset($sandbox['current'])) {
+    // The count of entities visited so far.
+    $sandbox['current'] = 0;
+    // Total entities that must be visited.
+    $sandbox['max'] = count($result);
+    // A place to store messages during the run.
+  }
+
+  // Get the first payment gateway for every payment module.
+  $payment_gateways_by_module = [];
+  /** @var \Drupal\commerce_payment\PaymentGatewayStorageInterface $payment_gateway_storage */
+  $payment_gateway_storage = \Drupal::entityTypeManager()->getStorage('commerce_payment_gateway');
+  $payment_gateways = $payment_gateway_storage->loadMultiple();
+  uasort($payment_gateways, [$payment_gateway_storage->getEntityType()->getClass(), 'sort']);
```

Comment:
You should be able to use $payment_gateway_storage->getEntityType()->getProvider() to get the module.